scrapers/utils/secrets.py

```diff
@@ -0,0 +1,50 @@
+import boto3
+import json
+import logging
+
+
+class SecretRetrievalError(Exception):
+    def __init__(self, message, secret_name=None, original_exception=None):
+        full_message = (
+            f"{message}. Secret: {secret_name}. Details: {original_exception}"
+            if original_exception
+            else message
+        )
+        super().__init__(full_message)
+
+        logging.error(full_message)
+        self.secret_name = secret_name
+        self.original_exception = original_exception
+
+
+def get_secret(secret_name, region="us-west-2"):
+    """
+    Retrieves secret value from a secret in AWS Secrets Manager.
+
+    Parameters:
+    - secret_name (str): The name of the secret.
+    - region (str): The AWS region where the secret is stored.
+
+    Returns:
+    - The value associated with the specified key.
+    """
+
+    client = boto3.client(service_name="secretsmanager", region_name=region)
+
+    try:
```

Comment:
This is looking better but we've got a lot of repetition now with the error handling.

We could simplify this to:
```
try:
    secret_response = client.get_secret_value(SecretId=secret_name)
    secret_string = secret_response.get("SecretString")
    if not secret_string:
        raise SecretRetrievalError("Secret is binary or unavailable in string format", secret_name)

    secret = json.loads(secret_string).get(secret_name)
    if not secret:
        raise SecretRetrievalError("The key was not found in the secret", secret_name)

    return secret

except Exception as e:
    raise SecretRetrievalError("Error retrieving secret", secret_name, e)

```

Reply:
Thanks for the suggestion! I incorporated this change into my code, tested it, reformatted it, and it works perfectly.